cs/src/Connections/RelayTunnelConnector.cs

```diff
@@ -193,6 +168,54 @@ public async Task ConnectSessionAsync(bool isReconnect, CancellationToken cancel
                     attempDelayMs <<= 1;
                 }
             }
+
+            async Task RefreshTunnelAccessTokenAsync(Exception exception)
+            {
+                var statusCode = exception.Data.Contains("HttpStatusCode") ? $" ({exception.Data["HttpStatusCode"]})" : string.Empty;
+                if (!isTunnelAccessTokenRefreshed)
+                {
+                    try
+                    {
+                        isTunnelAccessTokenRefreshed = await this.relayClient.RefreshTunnelAccessTokenAsync(cancellation);
+                    }
+                    catch (OperationCanceledException) when (cancellation.IsCancellationRequested)
+                    {
+                        throw;
+                    }
+                    catch (UnauthorizedAccessException uaex)
+                    {
+                        // The refreshed token is not valid.
+                        throw new UnauthorizedAccessException(
+                            $"Not authorized{statusCode}. Refreshed tunnel access token is not valid. {uaex.Message}",
```

Comment:
Add a space/parentheses: `$"Not authorized ({statusCode})`. Also below.

Reply:
It's added on line 174. If the status code is missing, `statusCode` will be and empty string. Otherwise, it is 
`$" ({exception.Data["HttpStatusCode"]})"`.

Reply:
Oh I see, I had missed the extra formatting above.